framework/src/media/FileInputDataSource.cpp

```diff
@@ -55,24 +55,37 @@ bool FileInputDataSource::open()
 {
 	if (!mFp) {
 		setAudioType(utils::getAudioTypeFromPath(mDataPath));
-
-		switch (getAudioType()) {
-		case AUDIO_TYPE_MP3:
-		case AUDIO_TYPE_AAC:
-		case AUDIO_TYPE_OPUS:
-			setDecoder(std::make_shared<Decoder>(getChannels(), getSampleRate()));
-			break;
-		case AUDIO_TYPE_FLAC:
-			/* To be supported */
-			break;
-		default:
-			/* Don't set any decoder for unsupported formats */
-			break;
-		}
+		unsigned int channel;
+		unsigned int sample_rate;
+		audio_format_type_t pcm_format;
 
 		mFp = fopen(mDataPath.c_str(), "rb");
 		if (mFp) {
-			medvdbg("file open success\n");
+			printf("file open success\n");
```

Comment:
If state including many lines but else is error case.
In this case, below style is good option.
```
if (!mp) {
   dbg();
   return false;
}
do_something
```